documentation/week3/src/App.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import './App.css';
+import Form from './components/Form';
+import List from './components/List';
+import data from './data/todos.json';
+import { loadFromLocalStorage, saveToLocalStorage } from './storage/localStoarge';
+export default class App extends Component {
+
+  state = loadFromLocalStorage() || {
+    todos: data,
+    id: 1,
+    discription: '',
+    deadline:''
+  }
+
+
+  handleDiscriptionChange = (event) => {
+    event.preventDefault();
+    this.setState({
+      discription: event.target.value
+    })
+  }
+
+  handleDeadlineChange = (event) => {
+    event.preventDefault();
+    this.setState({
+      deadline: event.target.value
+    })
+  }
+
+  handleClick = (id) => {
+    const todos = this.state.todos;
+    todos[id].done = !todos[id].done;
```

Comment:
By doing this you are mutating your state without using `this.setState`. We avoid mutating state because it will produce un-expected bugs also React will use its power of managing state using `this.setState`.

To understand this just write the following `console.log`:-

**Bad example**
```
const todos = this.state.todos;
    todos[id].done = !todos[id].done;
    console.log(todos[id]); // true
    console.log(this.state.todos[id]); // true
```
By doing this you will find that `this.state.todos[id]` already changed. However it shouldn't change until we use `this.setState`.

**Good example**
```
    const todos = this.state.todos.map( (todo) => (todo.id === id ? todo : {...todo, done: !todo.done}))
    console.log(todos[id]); // false
    console.log(this.state.todos[id]); // true
```
By deleting your code and just using maps, you will find that `this.state.todos[id]` didn't change and it will change only when you use `this.setState`.